x/gamm/keeper/msg_server_test.go

```diff
@@ -356,3 +362,171 @@ func (suite *KeeperTestSuite) TestExitPool_Events() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestMsgMigrateShares() {
+	defaultAccount := suite.TestAccs[0]
+	defaultGammShares := sdk.NewCoin("gamm/pool/1", sdk.MustNewDecFromStr("100000000000000000000").RoundInt())
+	invalidGammShares := sdk.NewCoin("gamm/pool/1", sdk.MustNewDecFromStr("100000000000000000001").RoundInt())
+	defaultAccountFunds := sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(200000000000)), sdk.NewCoin("usdc", sdk.NewInt(200000000000)))
+
+	type param struct {
+		sender                sdk.AccAddress
+		sharesToMigrateDenom  string
+		sharesToMigrateAmount sdk.Int
+		poolIdEntering        uint64
+	}
+
+	tests := []struct {
```

Comment:
We should check the exact state of the CL position that's created _somewhere_ (e.g. `expectedPosition`) even just as a sanity check so all our tests aren't relative to each other

Reply:
Agreed, added!

https://github.com/osmosis-labs/osmosis/pull/4051/commits/9efd06c8b20c8039c61ba2b0a16cd8529dcc14de